.github/workflows/ci.yml

```diff
@@ -38,27 +39,25 @@ jobs:
       - uses: actions/checkout@v4
       - name: Set up Python
         uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: "pip"
-      - name: Install dependencies
```

Comment:
this caching action was removed because it is not compatible with uv. A separate action can be added for uv caching. 

https://docs.astral.sh/uv/concepts/cache/